File: api/src/main/java/org/pmiops/workbench/actionaudit/auditors/AdminAuditor.java
package org.pmiops.workbench.actionaudit.auditors;

import org.pmiops.workbench.model.AccessReason;
import org.pmiops.workbench.model.AdminLockingRequest;

public interface AdminAuditor {
  void fireViewNotebookAction(
      String workspaceNamespace,
      String workspaceName,
      String notebookFilename,
      AccessReason accessReason);

  void fireLockWorkspaceAction(long workspaceId, AdminLockingRequest adminLockingRequest);

  void fireUnlockWorkspaceAction(long workspaceId);

  void firePublishWorkspaceAction(long workspaceId, String updatedCategory, String prevCategoryIfAny);

  void fireUnpublishWorkspaceAction(long workspaceId, String category);
}

File: api/src/main/java/org/pmiops/workbench/actionaudit/auditors/AdminAuditor.kt
package org.pmiops.workbench.actionaudit.auditors

import org.pmiops.workbench.model.AccessReason
import org.pmiops.workbench.model.AdminLockingRequest

interface AdminAuditor {
    fun fireViewNotebookAction(
        workspaceNamespace: String,
        workspaceName: String,
        notebookFilename: String,
        accessReason: AccessReason,
    )

    fun fireLockWorkspaceAction(
        workspaceId: Long,
        adminLockingRequest: AdminLockingRequest,
    )

    fun fireUnlockWorkspaceAction(workspaceId: Long)
}
